lib/testscript_engine/assertion.rb

```diff
@@ -137,11 +134,23 @@ def content_type(assert)
 
   def expression(assert)
     resource = get_resource(assert.sourceId)
+    raise AssertionException.new('Expression: No resource given by sourceId.', :fail) unless resource
+    raise AssertionException.new("Expression: Operator '#{assert.operator}' not supported in expression. Support only 'equals'", :fail) if assert.operator != nil && assert.operator != "equals"
+
+    received = evaluate_expression(assert.expression, resource)
+    expected = determine_expected_value(assert)
+    compare("Expression", received.to_s, assert.operator, expected.to_s)
+  end
+
+  def compare_to_source_id(assert)
+    resource = get_resource(assert.sourceId)
+    compare_to_source = get_resource(assert.compareToSourceId)
     raise AssertionException.new('No resource given by sourceId.', :fail) unless resource
+    raise AssertionException.new('No resource given by compareToSourceId.', :fail) unless compare_to_source
 
     received = evaluate_expression(assert.expression, resource)
     expected = determine_expected_value(assert)
-    compare("Expression", received, assert.operator, expected)
+    compare("CompareToSourceId", received.to_s, assert.operator, expected.to_s)
```

Comment:
The use of to_s here feels like a short cut that is probably right for now, but isn't always going to give you the correct results. For example, say you have a FHIRPath expression that evaluates to the boolean value `true` and another than evaluates to the string `"true"`, this logic won't be able to tell the difference. That said, I think ok to use this short-cut for now, but let's add an issue indicating that it is something to be resolved in the future by users interested in doing so.

Reply:
Good point. I think it's trade-off by converting everything into String, practically it support real testing cases, on the other hands it rather mess up testscript especially if testers not aware of the conversion enough. 

I'm okay with either keeping this conversion or remove to_s. Or put it as another configurability?